From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: roberth <roberth@arista.com>
Date: Wed, 24 Jan 2024 12:35:23 -0600
Subject: [PATCH] Disable warnings

When compiled with GCC, Insyde and AMD's C code spews out a lot of warnings. It
is what it is.

This patch disables all warnings by default to clean up the build output. If we
include our own custom modules in the build, we should remember to re-enable
these warnings.
---
 AMD/AmdEmbedded/AmdCpmPkg/Package.dsc | 1 +
 BaseTools/Conf/tools_def.template     | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/AMD/AmdEmbedded/AmdCpmPkg/Package.dsc b/AMD/AmdEmbedded/AmdCpmPkg/Package.dsc
index 8c194a4a..ad7972b4 100644
--- a/AMD/AmdEmbedded/AmdCpmPkg/Package.dsc
+++ b/AMD/AmdEmbedded/AmdCpmPkg/Package.dsc
@@ -504,3 +504,4 @@
   MSFT:RELEASE_*_*_CC_FLAGS            = /D AMD_TABLE_ADDITIONAL_BUFFER_SIZE=0x800
   RVCT:RELEASE_*_*_CC_FLAGS            = -DAMD_TABLE_ADDITIONAL_BUFFER_SIZE=0x800
   *_*_*_CC_FLAGS                       = -D AMD_TABLE_ADDITIONAL_BUFFER_SIZE=0x800
+  *_*_*_ASL_FLAGS                      = -ve
diff --git a/BaseTools/Conf/tools_def.template b/BaseTools/Conf/tools_def.template
index 94eed1c4..ee2f81d8 100644
--- a/BaseTools/Conf/tools_def.template
+++ b/BaseTools/Conf/tools_def.template
@@ -2741,7 +2741,7 @@ NOOPT_*_*_OBJCOPY_ADDDEBUGFLAG     = --add-gnu-debuglink=$(DEBUG_DIR)/$(MODULE_N
 *_*_*_DTCPP_PATH                   = DEF(DTCPP_BIN)
 *_*_*_DTC_PATH                     = DEF(DTC_BIN)
 
-DEFINE GCC_ALL_CC_FLAGS            = -g -Os -fshort-wchar -fno-builtin -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -include AutoGen.h -fno-common
+DEFINE GCC_ALL_CC_FLAGS            = -g -Os -fshort-wchar -fno-builtin -fno-strict-aliasing -w -Wno-array-bounds -include AutoGen.h -fno-common
 DEFINE GCC_IA32_CC_FLAGS           = DEF(GCC_ALL_CC_FLAGS) -m32  -mno-sse2 -mno-sse -mno-mmx -malign-double -freorder-blocks -freorder-blocks-and-partition -O2 -mno-stack-arg-probe
 DEFINE GCC_X64_CC_FLAGS            = DEF(GCC_ALL_CC_FLAGS) -mno-red-zone -Wno-address -mno-stack-arg-probe
 DEFINE GCC_ARM_CC_FLAGS            = DEF(GCC_ALL_CC_FLAGS) -mlittle-endian -mabi=aapcs -fno-short-enums -funsigned-char -ffunction-sections -fdata-sections -fomit-frame-pointer -Wno-address -mthumb -mfloat-abi=soft -fno-pic -fno-pie
@@ -4824,7 +4824,7 @@ RELEASE_ARMLINUXGCC_AARCH64_CC_FLAGS = $(ARCHCC_FLAGS) $(PLATFORM_FLAGS) DEF(GCC
 # VfrCompiler definitions
 ##################
 *_*_*_VFR_PATH                      = VfrCompile
-*_*_*_VFR_FLAGS                     = -l -n
+*_*_*_VFR_FLAGS                     = -l -n -vfrqidwarnoff
 
 ##################
 # VfcfCompiler definitions
-- 
2.41.0

